src/core/input_descriptor.rs

```diff
@@ -599,17 +599,13 @@ impl ConstraintsField {
         self.intent_to_retain
     }
 
-    /// Return the humanly-readable requested fields of the constraints field.
-    ///
-    /// This will convert camelCase to space-separated words with capitalized first letter.
+    /// Return the requested field in the format specified in the constraints field,
+    /// without changing its type casing, e.g. camelCase, snake_case, etc.
     ///
-    /// For example, if the path is `["dateOfBirth"]`, this will return `["Date of Birth"]`.
-    ///
-    /// This will also stripe the periods from the JSON path and return the last word in the path.
-    ///
-    /// e.g., `["$.verifiableCredential.credentialSubject.dateOfBirth"]` will return `["Date of Birth"]`.
-    /// e.g., `["$.verifiableCredential.credentialSubject.familyName"]` will return `["Family Name"]`.
+    /// This will stripe the delimiters from the JSON path and return the last value in the path.
     ///
+    /// e.g., `["$.verifiableCredential.credentialSubject.dateOfBirth"]` will return `["dateOfBirth"]`.
+    /// e.g., `["$.verifiableCredential.credentialSubject.familyName"]` will return `["familyName"]`.
     pub fn requested_fields(&self) -> Vec<String> {
         self.path()
```

Comment:
I'm actually wondering if we should just get rid of this function, as it only supports a very specific subset of JSON Path expressions. How would it handle inputs like:
```
$['verifiableCredential']['credentialSubject']['achievements'][*]['id']
$['verifiableCredential']['credentialSubject']['capabilities'][*]
$['verifiableCredential']['credentialSubject']['achievements'][?(@.evidence)]
```

Take a look at the `Example JSONPath Expressions` table [here](https://identity.foundation/presentation-exchange/spec/v2.0.0/#jsonpath-syntax-definition) for more examples.

Reply:
Yeah, I agree. We may wish to move this method closer to the parsed credential that matches the input descriptor.

After reviewing the JSONPath expressions, I don't think it belongs here either.